Library/Formula/vdirsyncer.rb

```diff
@@ -0,0 +1,62 @@
+class Vdirsyncer < Formula
+  desc "Synchronize calendars and contacts"
+  homepage "https://github.com/untitaker/vdirsyncer"
+  url "https://github.com/untitaker/vdirsyncer/archive/0.5.2.tar.gz"
+  sha256 "0992d4c0c98e71a0638050f3900125a82bd29e15d5be080a4520830fa48c660a"
+
+  option "without-keyring", "Build without python-keyring support"
+  depends_on :python3
+
+  if build.with? "keyring"
+    resource "keyring" do
+      url "https://pypi.python.org/packages/source/k/keyring/keyring-5.3.zip"
+      sha256 "ac2b4dc17e6edfb804b09ade15df79f251522e442976ea0c8ea0051474502cf5"
+    end
+  end
+
+  resource "click" do
+    url "https://pypi.python.org/packages/source/c/click/click-4.0.tar.gz"
+    sha256 "f49e03611f5f2557788ceeb80710b1c67110f97c5e6740b97edf70245eea2409"
+  end
+
+  resource "requests" do
+    url "https://pypi.python.org/packages/source/r/requests/requests-2.7.0.tar.gz"
+    sha256 "398a3db6d61899d25fd4a06c6ca12051b0ce171d705decd7ed5511517b4bb93d"
+  end
+
+  resource "lxml" do
+    url "https://pypi.python.org/packages/source/l/lxml/lxml-3.4.4.tar.gz"
+    sha256 "b3d362bac471172747cda3513238f115cbd6c5f8b8e6319bf6a97a7892724099"
+  end
+
+  resource "requests-toolbelt" do
+    url "https://pypi.python.org/packages/source/r/requests-toolbelt/requests-toolbelt-0.4.0.tar.gz"
+    sha256 "15b74b90a63841b8430d6301e5062cd92929b1074b0c95bf62166b8239db1a96"
+  end
+
+  resource "atomicwrites" do
+    url "https://pypi.python.org/packages/source/a/atomicwrites/atomicwrites-0.1.5.tar.gz"
+    sha256 "9b16a8f1d366fb550f3d5a5ed4587022735f139a4187735466f34cf4577e4eaa"
+  end
+
+  def install
+    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python3.4/site-packages"
+    %w[keyring click requests lxml requests-toolbelt atomicwrites].each do |r|
+      resource(r).stage do
+        system "python3", *Language::Python.setup_install_args(libexec/"vendor")
+      end
+    end
+
+    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python3.4/site-packages"
+    system "python3", *Language::Python.setup_install_args(libexec)
+
+    bin.install Dir["#{libexec}/bin/*"]
+    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
+  end
+
+  test do
+    ENV["LC_ALL"] = "en_US.UTF-8"
+    ENV["LANG"] = "en_US.UTF-8"
+    system "#{bin}/vdirsyncer", "--version"
```

Comment:
Again, we'd prefer actual usage rather than test scripts e.g. synchronise two folders whose data you populate in the `test do` block. Tests need to run in less than 30s.